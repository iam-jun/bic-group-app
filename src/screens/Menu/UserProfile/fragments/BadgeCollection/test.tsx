import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
