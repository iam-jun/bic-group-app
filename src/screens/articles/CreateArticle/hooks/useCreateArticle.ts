import { isEmpty, isEqual } from 'lodash';
import {
  useEffect, useMemo, useRef, useState,
} from 'react';
import { Keyboard } from 'react-native';

import moment from 'moment';
import { useShallow } from '~/store/utils';
import useMentionInputStore from '~/beinComponents/inputs/MentionInput/store';
import IMentionInputState from '~/beinComponents/inputs/MentionInput/store/Interface';
import {
  IEditArticleAudience,
  IEditArticleData,
  IParamsValidateSeriesTags,
  IPayloadPublishDraftArticle,
  IPayloadPutEditArticle,
} from '~/interfaces/IArticle';
import { withNavigation } from '~/router/helper';
import { getAudienceIdsFromAudienceObject, isEmptyContent } from '~/screens/articles/CreateArticle/helper';
import useCreateArticleStore from '~/screens/articles/CreateArticle/store';
import { getMentionsFromContent } from '~/helpers/post';
import usePostsStore from '~/store/entities/posts';
import postsSelector from '~/store/entities/posts/selectors';
import { useBaseHook } from '~/hooks';

import { rootNavigationRef } from '~/router/refs';
import articleStack from '~/router/navigator/MainStack/stacks/articleStack/stack';
import useDraftArticleStore from '~/screens/Draft/DraftArticle/store';
import useScheduleArticlesStore from '~/screens/YourContent/components/ScheduledArticles/store';
import useModalStore from '~/store/modal';
import { PostStatus } from '~/interfaces/IPost';

interface IHandleSaveOptions {
  isShowLoading?: boolean;
  isNavigateBack?: boolean;
  isShowToast?: boolean;
  shouldValidateSeriesTags?: boolean;
  onSuccess?: () => void;
  titleAlert?: string;
}

const navigation = withNavigation(rootNavigationRef);

export interface IUseEditArticle {
  articleId: string;
  handleSaveAudienceError?: (listIdAudiences: string[]) => void;
}

const useCreateArticle = ({
  articleId,
}: IUseEditArticle) => {
  const article = usePostsStore(postsSelector.getPost(articleId, {}));

  const actions = useCreateArticleStore((state) => state.actions);

  const data = useCreateArticleStore((state) => state.data, useShallow) || {};
  const loading = useCreateArticleStore((state) => state.loading);
  const isValidating = useCreateArticleStore((state) => state.isValidating);
  const isDraft = useCreateArticleStore((state) => state.isDraft);
  const publishedAt = useCreateArticleStore((state) => state.schedule.publishedAt);

  const { showAlert } = useModalStore((state) => state.actions);

  const [isShowToastAutoSave, setShowToastAutoSave] = useState<boolean>(false);

  const refStopTyping = useRef(null);
  const refTypingConstantly = useRef(null);

  const tempMentions = useMentionInputStore(
    (state: IMentionInputState) => state.tempSelected,
  );

  const groupIds = useMemo(
    () => data.audience?.groupIds?.join?.(','),
    [data.audience],
  );

  const { t } = useBaseHook();

  const isValidScheduleTime = () => moment(publishedAt).isSameOrAfter(moment());

  // auto save for draft article, so no need to check if content is empty
  const isDraftContentUpdated
    = article.content !== data.content;

  const isHasChange = () => {
    // self check at src/screens/articles/CreateArticle/screens/CreateArticleContent/index.tsx
    // const isContentUpdated = article.content !== data.content && !isEmptyContent();
    const isSummaryUpdated = article.summary !== data.summary;
    const isTitleUpdated = article.title !== data.title;
    const isCategoriesUpdated
      = !isEqual(article.categories, data.categories);
    // self check at src/screens/articles/CreateArticle/screens/CreateArticleAudience/index.tsx
    // const isAudienceUpdated = !isEqual(getAudienceIdsFromAudienceObject(article.audience), data.audience)
    // && !(isEmpty(data.audience?.groupIds) && isEmpty(data.audience?.userIds));
    const isCoverMediaUpdated
      = article.coverMedia?.id !== data.coverMedia?.id;
    const isSeriesUpdated = !isEqual(article?.series, data.series);
    const isTagsUpdated = !isEqual(article?.tags, data.tags);

    return (
      isTitleUpdated
      || isSummaryUpdated
      || isCategoriesUpdated
      || isCoverMediaUpdated
      || isSeriesUpdated
      || isTagsUpdated
    );
  };

  const isValidForSave = () => {
    // self check at src/screens/articles/CreateArticle/screens/CreateArticleContent/index.tsx
    // const isContentUpdated = article.content !== data.content && !isEmptyContent();
    const isSummaryUpdated = article.summary !== data.summary;
    const isTitleUpdated = article.title !== data.title && !isEmpty(data.title?.trim());
    const isCategoriesUpdated
      = !isEqual(article.categories, data.categories)
      && !isEmpty(data.categories);
    // self check at src/screens/articles/CreateArticle/screens/CreateArticleAudience/index.tsx
    // const isAudienceUpdated = !isEqual(getAudienceIdsFromAudienceObject(article.audience), data.audience)
    // && !(isEmpty(data.audience?.groupIds) && isEmpty(data.audience?.userIds));
    const isCoverMediaUpdated
      = article.coverMedia?.id !== data.coverMedia?.id
      && !isEmpty(data.coverMedia);
    const isSeriesUpdated = !isEqual(article?.series, data.series);
    const isTagsUpdated = !isEqual(article?.tags, data.tags);

    return (
      isTitleUpdated
      || isSummaryUpdated
      || isCategoriesUpdated
      || isCoverMediaUpdated
      || isSeriesUpdated
      || isTagsUpdated
    );
  };

  const getValidButtonPublish = () => {
    const isTitleValid = !isEmpty(data.title);
    const isContentValid = !isEmptyContent(data.content);
    const isCategoriesValid = !isEmpty(data.categories);
    const isCoverValid = !isEmpty(data.coverMedia);
    const isAudienceValid = !(
      isEmpty(data.audience?.groupIds) && isEmpty(data.audience?.userIds)
    );

    return (
      isTitleValid
      && isContentValid
      && isCategoriesValid
      && isCoverValid
      && isAudienceValid
    );
  };

  const initEditStoreData = () => {
    const {
      id,
      title,
      content,
      audience: audienceObject,
      mentions,
      summary,
      categories,
      coverMedia,
      series,
      tags,
      status,
      publishedAt,
    } = article;
    const audienceIds: IEditArticleAudience
      = getAudienceIdsFromAudienceObject(audienceObject);
    const data: IEditArticleData = {
      id,
      title,
      content,
      audience: audienceIds,
      mentions,
      summary,
      categories,
      coverMedia,
      series,
      tags,
    };
    actions.setData(data);
    const isDraft = [PostStatus.DRAFT, PostStatus.WAITING_SCHEDULE, PostStatus.SCHEDULE_FAILED].includes(status);
    actions.setIsDraft(isDraft);
    if (isDraft) {
      actions.setPublishedAt(publishedAt || '');
    }
  };

  useEffect(() => {
    if (!isEmpty(article)) {
      initEditStoreData();
    }
  }, [article]);

  const enableButtonSave = isValidForSave();
  const validButtonPublish = getValidButtonPublish();
  const isChanged = isHasChange();

  const handleContentChange = (newContent: string) => {
    actions.setContent(newContent);
  };

  const handleTitleChange = (newTitle: string) => {
    actions.setTitle(newTitle);
  };

  const resetPublishedAt = () => {
    actions.setPublishedAt(article?.publishedAt || '');
  };

  const updateMentions = () => {
    const newMentions = getMentionsFromContent(data.content, tempMentions);
    actions.setMentions(newMentions);
  };

  const handleAudiencesChange = (newAudiences: IEditArticleAudience) => {
    actions.setAudience(newAudiences);
  };

  const showToastAutoSave = () => {
    setShowToastAutoSave(true);
    setTimeout(() => {
      setShowToastAutoSave(false);
    }, 2000);
  };

  const debounceStopTyping = () => {
    clearTimeout(refStopTyping.current);
    refStopTyping.current = setTimeout(() => {
      clearTimeout(refTypingConstantly.current);
      refTypingConstantly.current = null;
      handleSave({ isNavigateBack: false, isShowLoading: false, isShowToast: false });
      showToastAutoSave();
    }, 500);
  };

  const debounceTypingConstantly = () => {
    if (!refTypingConstantly.current) {
      refTypingConstantly.current = setTimeout(() => {
        handleSave({ isNavigateBack: false, isShowLoading: false, isShowToast: false });
        refTypingConstantly.current = null;
        showToastAutoSave();
      }, 5000);
    }
  };

  useEffect(() => {
    // only auto save for draft article
    if (isDraft && isDraftContentUpdated) {
      debounceStopTyping();
      debounceTypingConstantly();
    }
  }, [data.content]);

  useEffect(() => () => {
    actions.setIsDraft(false);
  }, []);

  const validateSeriesTags = (onSuccess: (response) => void, onError: (error) => void) => {
    const dataUpdate = useCreateArticleStore.getState().data;
    const validateParams: IParamsValidateSeriesTags = {
      groups: dataUpdate?.audience?.groupIds || [],
      series: dataUpdate?.series?.map?.((item) => item.id) || [],
      tags: dataUpdate?.tags?.map?.((item) => item.id) || [],
    };
    actions.validateSeriesTags(validateParams, onSuccess, onError);
  };

  const handleSave = (options?: IHandleSaveOptions) => {
    Keyboard.dismiss();
    const {
      isNavigateBack = true, isShowLoading = true, isShowToast = true, shouldValidateSeriesTags, onSuccess, titleAlert,
    } = options || {};
    updateMentions();

    // get data directly from store instead of hook
    // because in case we set store and immediately call handleSave
    // useCreateArticle hook is still not updated at that time
    // so handleSave will hold the old data instead of the new data
    const dataUpdate = useCreateArticleStore.getState().data;
    const putEditArticleParams = {
      articleId,
      data: dataUpdate,
      isNavigateBack,
      isShowToast,
      isShowLoading,
      onSuccess,
    } as IPayloadPutEditArticle;

    if (shouldValidateSeriesTags) {
      const onSuccess = () => actions.putEditArticle(putEditArticleParams);
      const onError = (error) => {
        actions.handleSaveError(error, () => handleSave(options), titleAlert);
      };
      validateSeriesTags(onSuccess, onError);
    } else {
      actions.putEditArticle(putEditArticleParams);
    }
  };

  const handlePublish = () => {
    if (!validButtonPublish) return;

    const goToArticleDetail = () => {
      navigation.replace(articleStack.articleDetail, { articleId: data.id });
    };

    const onHandleSaveErrorDone = () => handleSave({
      isNavigateBack: false,
      isShowLoading: true,
      isShowToast: false,
      onSuccess: () => handlePublish(),
    });

    const payload: IPayloadPublishDraftArticle = {
      draftArticleId: data.id,
      onSuccess: () => {
        goToArticleDetail();
        useScheduleArticlesStore.getState().actions.getScheduleArticles({ isRefresh: true });
      },
      onError: (error) => actions.handleSaveError(error, onHandleSaveErrorDone),
    };
    useDraftArticleStore.getState().actions.publishDraftArticle(payload);
  };

  const handleSchedule = () => {
    if (!validButtonPublish) return;

    if (!isValidScheduleTime()) {
      actions.setErrorScheduleSubmiting(t('article:fail_schedule'));
      return;
    }

    actions.scheduleArticle();
  };

  const handleBack = (shouldShowAlert = false) => {
    if (isChanged || shouldShowAlert) {
      Keyboard.dismiss();
      showAlert({
        title: t('discard_alert:title'),
        content: t('discard_alert:content'),
        cancelBtn: true,
        cancelLabel: t('common:btn_discard'),
        confirmLabel: t('common:btn_stay_here'),
        onCancel: () => {
          initEditStoreData();
          navigation.goBack();
        },
      });
      return;
    }
    navigation.goBack();
  };

  return {
    loading,
    isValidating,
    isShowToastAutoSave,
    enableButtonSave,
    validButtonPublish,
    title: data.title,
    content: data.content,
    groupIds,
    handleTitleChange,
    handleContentChange,
    handleSave,
    handleBack,
    handleAudiencesChange,
    handlePublish,
    handleSchedule,
    validateSeriesTags,
    resetPublishedAt,
  };
};

export default useCreateArticle;
