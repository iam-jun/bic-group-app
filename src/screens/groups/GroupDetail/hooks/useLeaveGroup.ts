import { useDispatch } from 'react-redux';
import i18next from 'i18next';

import modalActions from '~/storeRedux/modal/actions';
import Text from '~/baseComponents/Text';
import { handleLeaveInnerGroups } from '../../helper';
import useGroupDetailStore from '../store';
import { GroupPrivacyType } from '~/constants/privacyTypes';

interface IUseLeaveGroup {
  groupId: string;
  username: string;
  privacy: GroupPrivacyType;
}

const useLeaveGroup = ({ groupId, username, privacy }: IUseLeaveGroup) => {
  const dispatch = useDispatch();
  const actions = useGroupDetailStore((state) => state.actions);

  const doLeaveGroup = () => {
    actions.leaveGroup(groupId, privacy);
  };

  const alertPayload = {
    iconName: 'ArrowRightFromArc',
    title: i18next.t('groups:modal_confirm_leave_group:title'),
    content: i18next.t('groups:modal_confirm_leave_group:description'),
    ContentComponent: Text.BodyS,
    cancelBtn: true,
    onConfirm: doLeaveGroup,
    confirmLabel: i18next.t('groups:modal_confirm_leave_group:button_leave'),
  };

  const getInnerGroupsText = (innerGroups: any) => {
    if (innerGroups.length > 0) {
      alertPayload.content
        += ` ${i18next.t('groups:modal_confirm_leave_group:leave_inner_groups')}`;

      const groupsLeaveToString = innerGroups.join(', ');
      alertPayload.content = alertPayload.content.replace(
        '{0}',
        groupsLeaveToString,
      );
    }
    dispatch(modalActions.showAlert(alertPayload));
  };

  const alertLeaveGroup = () => {
    handleLeaveInnerGroups(
      groupId, username, dispatch, getInnerGroupsText,
    );
  };

  return alertLeaveGroup;
};

export default useLeaveGroup;
