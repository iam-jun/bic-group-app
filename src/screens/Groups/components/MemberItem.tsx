import {StyleSheet} from 'react-native';
import React from 'react';

import Text from '~/beinComponents/Text';
import {ITheme} from '~/theme/interfaces';
import PrimaryItem from '~/beinComponents/list/items/PrimaryItem';
import Icon from '~/beinComponents/Icon';
import {useTheme} from 'react-native-paper';
import images from '~/resources/images';
import useAuth from '~/hooks/auth';
import {formatDMLink} from '~/utils/link';
import {openLink} from '~/utils/common';
import {useRootNavigation} from '~/hooks/navigation';
import mainStack from '~/router/navigator/MainStack/stack';
import {useKeySelector} from '~/hooks/selector';
import groupsKeySelector from '../redux/keySelector';

interface MemberItemProps {
  item: any;
  canManageMember: boolean;
  onPressMenu: (item: any) => void;
}

const MemberItem = ({item, canManageMember, onPressMenu}: MemberItemProps) => {
  const theme = useTheme() as ITheme;
  const styles = createStyles(theme);
  const {colors} = theme;
  const {user} = useAuth();
  const communityDetail = useKeySelector(groupsKeySelector.communityDetail);
  const {rootNavigation} = useRootNavigation();

  const {id, fullname, avatar, username} = item || {};

  const goToUserProfile = () => {
    rootNavigation.navigate(mainStack.userProfile, {userId: id});
  };

  const onPressChat = () => {
    if (!username) return;
    const link = formatDMLink(communityDetail.slug, username);
    openLink(link);
  };

  return (
    <PrimaryItem
      showAvatar
      testID="member_item"
      style={styles.itemContainer}
      avatar={avatar || images.img_user_avatar_default}
      avatarProps={{isRounded: true, variant: 'medium'}}
      onPress={goToUserProfile}
      ContentComponent={
        <Text.Body numberOfLines={1}>
          {fullname}
          <Text.BodyS
            color={colors.textSecondary}>{` @${username}`}</Text.BodyS>
        </Text.Body>
      }
      RightComponent={
        <>
          {user?.username !== username && (
            <Icon
              icon={'MessageDots'}
              backgroundColor={colors.bgSecondary}
              style={styles.iconChat}
              onPress={onPressChat}
              buttonTestID="member_item.icon_chat.button"
            />
          )}
          {canManageMember && (
            <Icon
              icon={'EllipsisV'}
              style={styles.iconOption}
              onPress={() => onPressMenu(item)}
              buttonTestID="member_item.icon_option.button"
            />
          )}
        </>
      }
    />
  );
};

const createStyles = (theme: ITheme) => {
  const {spacing} = theme;

  return StyleSheet.create({
    itemContainer: {
      height: undefined,
      paddingHorizontal: spacing.padding.large,
      paddingVertical: spacing.padding.tiny,
    },
    iconChat: {
      height: 36,
      width: 36,
      justifyContent: 'center',
      alignItems: 'center',
      padding: spacing.padding.small,
      marginLeft: spacing.margin.tiny,
    },
    iconOption: {
      marginLeft: spacing.margin.small,
    },
  });
};

export default MemberItem;
