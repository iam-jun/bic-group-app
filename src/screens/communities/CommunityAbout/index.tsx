import React from 'react';
import Header from '~/beinComponents/Header';
import ScreenWrapper from '~/beinComponents/ScreenWrapper';
import { useBaseHook } from '~/hooks';
import { useRootNavigation } from '~/hooks/navigation';
import { useKeySelector } from '~/hooks/selector';
import groupStack from '~/router/navigator/MainStack/stacks/groupStack/stack';
import groupsKeySelector from '~/storeRedux/groups/keySelector';
import AboutContent from '../CommunityDetail/components/AboutContent';
import groupJoinStatus from '~/constants/groupJoinStatus';

const CommunityAbout = () => {
  const { rootNavigation } = useRootNavigation();
  const { t } = useBaseHook();
  const infoDetail = useKeySelector(groupsKeySelector.communityDetail);
  const { name, id, joinStatus } = infoDetail;
  const isMember = joinStatus === groupJoinStatus.member;

  const onPressTotalMember = () => {
    rootNavigation.navigate(groupStack.communityMembers, { communityId: id, isMember });
  };

  return (
    <ScreenWrapper isFullView>
      <Header title={`${t('settings:title_about')} ${name}`} />
      <AboutContent profileInfo={infoDetail} onPressMember={onPressTotalMember} />
    </ScreenWrapper>
  );
};

export default CommunityAbout;
