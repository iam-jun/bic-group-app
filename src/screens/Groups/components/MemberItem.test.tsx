import React from 'react';

import MemberItem from './MemberItem';
import initialState from '~/store/initialState';
import { renderWithRedux, createTestStore, fireEvent } from '~/test/testUtils';
import { adminDetail } from '~/test/mock_data/communities';
import * as navigationHook from '~/hooks/navigation';
import mainStack from '~/router/navigator/MainStack/stack';

describe('MemberItem component', () => {
  const canManageMember = true;
  const onPressMenu = jest.fn();

  it('should render data correctly', () => {
    const state = { ...initialState };
    state.auth.user = { username: 'test' };
    const item = { ...adminDetail };
    const store = createTestStore(state);
    const wrapper = renderWithRedux(
      <MemberItem
        item={item}
        canManageMember={canManageMember}
        onPressMenu={onPressMenu}
      />,
      store,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render icon member option correctly when user is an admin', () => {
    const state = { ...initialState };
    state.auth.user = { username: 'anothertest' };
    const item = { ...adminDetail };
    const store = createTestStore(state);
    const wrapper = renderWithRedux(
      <MemberItem
        item={item}
        canManageMember={canManageMember}
        onPressMenu={onPressMenu}
      />,
      store,
    );
    const iconOption = wrapper.getByTestId('member_item.icon_option.button');
    expect(iconOption).toBeDefined();
    fireEvent.press(iconOption);
    expect(wrapper).toMatchSnapshot();
  });

  it('should NOT render icon member option correctly when user is NOT an admin', () => {
    const state = { ...initialState };
    state.auth.user = { username: 'anothertest' };
    const item = { ...adminDetail };
    const store = createTestStore(state);
    const wrapper = renderWithRedux(
      <MemberItem
        item={item}
        canManageMember={false}
        onPressMenu={onPressMenu}
      />,
      store,
    );
    const iconClose = wrapper.queryByTestId('member_item.icon_option.button');
    expect(iconClose).toBeNull();
  });

  it('should navigate to User profile correctly', () => {
    const state = { ...initialState };
    state.auth.user = { username: 'test' };
    const item = { ...adminDetail };
    const store = createTestStore(state);
    const navigate = jest.fn();
    const rootNavigation = { navigate };
    jest.spyOn(navigationHook, 'useRootNavigation').mockImplementation(() => ({ rootNavigation } as any));

    const wrapper = renderWithRedux(
      <MemberItem
        item={item}
        canManageMember={canManageMember}
        onPressMenu={onPressMenu}
      />,
      store,
    );

    const memberItem = wrapper.getByTestId('member_item');
    fireEvent.press(memberItem);
    expect(navigate).toBeCalledWith(mainStack.userProfile, {
      userId: adminDetail.id,
    });
  });
});
