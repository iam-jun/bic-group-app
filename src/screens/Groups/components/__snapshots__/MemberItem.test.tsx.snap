// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberItem component should render data correctly 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": undefined,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 4,
      }
    }
    testID="member_item"
  >
    <View
      style={
        Object {
          "borderRadius": 4,
          "height": 40,
          "marginRight": 12,
          "width": 40,
        }
      }
      testID="avatar"
    >
      <View
        style={
          Object {
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
        testID="avatar_container"
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "position": "relative",
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f9f9f9",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                }
              }
              testID="Image__placeholder"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                },
              ]
            }
          >
            <FastImageView
              onFastImageError={[Function]}
              onFastImageLoadEnd={[Function]}
              resizeMode="cover"
              source={
                Object {
                  "uri": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/e41d70bc-0bfd-4c56-a461-c2dfb10434c1.jpg",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="avatar.image"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#181C32",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 15,
            },
            Object {},
            undefined,
          ]
        }
      >
        Test Name
        <Text
          style={
            Array [
              Object {
                "color": "#181C32",
                "fontFamily": "BeVietnamPro-Light",
                "fontSize": 13,
              },
              Object {
                "color": "#767D93",
              },
              undefined,
            ]
          }
        >
           @test
        </Text>
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="member_item.icon_option.button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "marginLeft": 8,
            },
            Object {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <View />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`MemberItem component should render icon member option correctly when user is an admin 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": undefined,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 4,
      }
    }
    testID="member_item"
  >
    <View
      style={
        Object {
          "borderRadius": 4,
          "height": 40,
          "marginRight": 12,
          "width": 40,
        }
      }
      testID="avatar"
    >
      <View
        style={
          Object {
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
        testID="avatar_container"
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "position": "relative",
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f9f9f9",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                }
              }
              testID="Image__placeholder"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                },
              ]
            }
          >
            <FastImageView
              onFastImageError={[Function]}
              onFastImageLoadEnd={[Function]}
              resizeMode="cover"
              source={
                Object {
                  "uri": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/e41d70bc-0bfd-4c56-a461-c2dfb10434c1.jpg",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="avatar.image"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#181C32",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 15,
            },
            Object {},
            undefined,
          ]
        }
      >
        Test Name
        <Text
          style={
            Array [
              Object {
                "color": "#181C32",
                "fontFamily": "BeVietnamPro-Light",
                "fontSize": 13,
              },
              Object {
                "color": "#767D93",
              },
              undefined,
            ]
          }
        >
           @test
        </Text>
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="member_item.icon_chat.button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
              "height": 36,
              "justifyContent": "center",
              "marginLeft": 4,
              "padding": 8,
              "width": 36,
            },
            Object {
              "backgroundColor": "#F1F2F8",
            },
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="member_item.icon_option.button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "marginLeft": 8,
            },
            Object {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <View />
        </View>
      </View>
    </View>
  </View>
</View>
`;
